### DISCUSSION TITLE
如何實現 udp 80 -> 443 端口轉發？

### DISCUSSION THREAD

**Discussion by LearZhou** [XTLS/Xray-core#1977] [Q&A]
根據 R 大對 Reality 的描述，最好在服務器端要具有 udp 80 -> 443 端口轉發的能力，對這一點我挺疑惑的，請教各位高手、高高手。

tcp 的 80 -> 443 轉發很熟悉了，用反向代理軟件 nginx、HaProxy 之類的都可以實現，但對 udp 的相應端口轉發，請問如何實現呢？

**Comment by nursery01** [comment]
linux版的nginx也可以用於udp

windows版的nginx只支持tcp

**Comment by ghost** [comment]
转发 udp 80 端口干什么？转发 443 udp 就行了。就算不转发 udp 443 在很多情况也不是什么强特征

**Comment by chika0801** [comment]
cat > /etc/sysctl.d/ip_forward.conf << EOF
net.ipv4.ip_forward=1
EOF
sysctl -p /etc/sysctl.d/ip_forward.conf

apt install -y iptables-persistent


添加并保存
iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 80 -j DNAT --to-destination 目标网址的IP:80
iptables -t nat -A PREROUTING -i eth0 -p udp --dport 443 -j DNAT --to-destination 目标网址的IP:443
iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
netfilter-persistent save

查看
iptables -t nat -nL --line

删除并保存
iptables -t nat -D PREROUTING 1
iptables -t nat -D PREROUTING 1
iptables -t nat -D POSTROUTING 1
netfilter-persistent save


问的chatgpt
